92-AppendixC.Rmd

```diff
@@ -0,0 +1,494 @@
+
+* Stata has "extended" missing values, .A through .Z.
+* SAS has "special" missing values, .A through .Z and ._.
+* SPSS has per-column "user" missing values. Each column can declare up to three distinct values or a range of values (plus one distinct value) that should be treated as missing.
+
+SAS and Stata use a concept known as 'tagged' missing values, which extend R's regular `NA`. A 'tagged' missing value is essentially an `NA` with an additional single-character label. These values behave identically to regular `NA` in standard R operations while preserving the informative tag associated with the missing value.
+
+Here is an example from the NORC at the University of Chicago’s 2018 General Society Survey.
+
+```r
+head(gss_dta$HEALTH)
+#> <labelled<double>[6]>: condition of health
+#> [1]     2     1 NA(i) NA(i)     1     2
+#> 
+#> Labels:
+#>  value     label
+#>      1 excellent
+#>      2      good
+#>      3      fair
+#>      4      poor
+#>  NA(d)        DK
+#>  NA(i)       IAP
+#>  NA(n)        NA
+```
+
+In contrast, SPSS uses a different approach called 'user-defined values' to denote missing values. Each column in an SPSS dataset can have up to three distinct values designated as missing or a specified range of missing values. To model these additional user-defined missing values, {haven} provides the `labeled_spss()` subclass of `labeled()`. When you import SPSS data using {haven}, it ensures that user-defined missing values are correctly handled. You can work with this data in R while preserving the unique missing value conventions from SPSS.
+
+Here is what the GSS SPSS data looks like when loaded with {haven}.
+
+```
+head(gss_sps$HEALTH)
+#> <labelled_spss<double>[6]>: Condition of health
+#> [1] 2 1 0 0 1 2
+#> Missing values: 0, 8, 9
+#> 
+#> Labels:
+#>  value     label
+#>      0       IAP
+#>      1 EXCELLENT
+#>      2      GOOD
+#>      3      FAIR
+#>      4      POOR
+#>      8        DK
+#>      9        NA
+```
+
+## Importing data from APIs into R
+
+In addition to working with data saved as files, we may also need to retrieve data through Application Programming Interfaces (APIs). APIs provide a structured way to access data hosted on external servers and import it directly into R for analysis.
+
+To access this data, you need to understand how to construct API requests. Each API has unique endpoints, parameters, and authentication requirements. Pay attention to:
+
+* Endpoints: These are URLs that point to specific data or services.
+* Parameters: Information you pass to the API to customize your request (e.g., date ranges, filters).
+* Authentication: APIs may require API keys or tokens for access.
+* Rate Limits: APIs may have usage limits, so be aware of any rate limits or quotas.
+
+Typically, we begin by making a GET request to an API endpoint. The {httr2} package allows us to generate and process HTTP requests. We can make the GET request by pointing to the URL that contains the data we would like.
```

Comment:
Is there a reason why GET is all caps?  Does this stand for something?  Are we using GET because that's the name of the function in R?